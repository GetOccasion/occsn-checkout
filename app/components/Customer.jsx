import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

import { Input } from 'mitragyna';
import { InputGroup } from 'reactstrap';

import occsn from '../libs/Occasion';

export default class Customer extends PureComponent {
  static propTypes = {
    subject: PropTypes.instanceOf(occsn.Customer),
  };

  render() {
    let { subject } = this.props;

    return <InputGroup>
      <Input type="email" name="email" className="form-control"></Input>
      <Input type="text" name="firstName" className="form-control"></Input>
      <Input type="text" name="lastName" className="form-control"></Input>
      <Input type="text" name="zip" className="form-control"></Input>
    </InputGroup>;
  }
}
