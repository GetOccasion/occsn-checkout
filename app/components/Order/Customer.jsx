import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

import { ErrorsFor, Field } from 'mitragyna';
import { FormFeedback, Input, ListGroup, ListGroupItem } from 'reactstrap';

import occsn from '../../libs/Occasion';

export default class Customer extends PureComponent {
  static propTypes = {
    subject: PropTypes.instanceOf(occsn.Customer),
  };

  render() {
    let { subject } = this.props;

    return <section>
      <ListGroup>
        <ListGroupItem>
          <Field type="email" name="email" placeholder="Email" component={ Input } invalidClassName='is-invalid'></Field>
          <ErrorsFor component={FormFeedback} field='email'></ErrorsFor>
        </ListGroupItem>
        <ListGroupItem>
          <Field type="text"  name="firstName" placeholder="First Name" component={ Input } invalidClassName='is-invalid'></Field>
          <ErrorsFor component={FormFeedback} field='firstName'></ErrorsFor>
        </ListGroupItem>
        <ListGroupItem>
          <Field type="text"  name="lastName" placeholder="Last Name" component={ Input } invalidClassName='is-invalid'></Field>
          <ErrorsFor component={FormFeedback} field='lastName'></ErrorsFor>
        </ListGroupItem>
        <ListGroupItem>
          <Field type="text"  name="zip" placeholder="Zip Code" component={ Input } invalidClassName='is-invalid'></Field>
          <ErrorsFor component={FormFeedback} field='zip'></ErrorsFor>
        </ListGroupItem>
      </ListGroup>
    </section>;
  }
}
