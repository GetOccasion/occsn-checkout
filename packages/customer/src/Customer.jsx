import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'

import classnames from 'classnames';
import { ErrorsFor, Field, Resource } from 'mitragyna'
import { FormFeedback, FormGroup, Input, Label } from 'reactstrap'

import {OccsnContext} from '@occsn/occsn-provider'

export class Customer extends PureComponent {
  static contextType = OccsnContext

  constructor() {
    super();

    this.state = {}
  }

  render() {
    const { className } = this.props
    const { components: { order }} = this.context

    let classNames = classnames('occsn-customer', className)

    let customer = order.getOrder().customer()

    return (
      <section className={classNames}>
        <Resource reflection="customer" subject={customer}>
          <FormGroup className="customer-email-form-group">
            <Label for="email">Email*</Label>
            <Field
              className="customer-email-field"
              type="email"
              name="email"
              id="email"
              component={Input}
              invalidClassName="is-invalid"
            />
            <ErrorsFor className="customer-email-errors" component={FormFeedback} field="email" />
          </FormGroup>
          <FormGroup className="customer-first-name-form-group">
            <Label for="firstName">First Name*</Label>
            <Field
              className="customer-first-name-field"
              type="text"
              name="firstName"
              id="firstName"
              component={Input}
              invalidClassName="is-invalid"
            />
            <ErrorsFor
              className="customer-first-name-errors"
              component={FormFeedback}
              field="firstName"
            />
          </FormGroup>
          <FormGroup className="customer-last-name-form-group">
            <Label for="lastName">Last Name*</Label>
            <Field
              className="customer-last-name-field"
              type="text"
              name="lastName"
              id="lastName"
              component={Input}
              invalidClassName="is-invalid"
            />
            <ErrorsFor
              className="customer-last-name-errors"
              component={FormFeedback}
              field="lastName"
            />
          </FormGroup>
          <FormGroup className="customer-zip-form-group">
            <Label for="zip">Zip Code*</Label>
            <Field
              className="customer-zip-field"
              type="text"
              name="zip"
              id="zip"
              component={Input}
              invalidClassName="is-invalid"
            />
            <ErrorsFor className="customer-zip-errors" component={FormFeedback} field="zip" />
          </FormGroup>
        </Resource>
      </section>
    )
  }
}
