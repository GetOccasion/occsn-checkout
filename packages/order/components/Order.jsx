import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'

import classnames from 'classnames';
import { Resource } from 'mitragyna'
import { Container, Row, Col } from 'reactstrap'

import OccsnContextConsumer from '@occsn/occsn-provider';

export class Order extends PureComponent {
  static contextType = OccsnContextConsumer;

  static propTypes = {
    product: PropTypes.object.isRequired
  }

  constructor({ product }) {
    const { Product } = this.context;
    if(!product.isA?(Product)) {
      throw TypeError('Order component received prop product not of type Product');
    }

    this.state = {};
  }

  componentDidMount() {
    this.constructOrder();
  }

  async bookOrder(order) {
    try {
      order = await order.update({ status: 'reserved' })
    } catch (invalidOrder) {
      order = invalidOrder
    }

    this.setOrder(order)
  }

  constructOrder() {
    const { product } = this.props;
    const { Order } = this.context;

    occsn.Order.construct({ product, status: 'initialized' }).then(this.setOrder);
  }

  async saveOrder(order) {
    this.setOrder(await order.save());
  }

  setOrder(order) {
    this.setState({ order });
  }

  render() {
    const { children, className } = this.props;
    const { order } = this.state;

    let classNames = classnames('occsn-order-form', className);

    return (
      <section className={classNames}>
        {order && (
          <Resource
            afterError={this.setOrder}
            afterUpdate={this.saveOrder}
            onInvalidSubmit={this.setOrder}
            onSubmit={this.bookOrder}
            subject={order}
          >
            {children}
          </Resource>
        )}
      </section>
    )
  }
}
