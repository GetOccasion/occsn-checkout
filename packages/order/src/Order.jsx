import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'

import classnames from 'classnames';
import { Resource } from 'mitragyna'
import { Container, Row, Col } from 'reactstrap'
import _ from 'underscore';

import {OccsnContext} from '@occsn/occsn-provider';

export default class Order extends PureComponent {
  static contextType = OccsnContext

  static propTypes = {
    onChange: PropTypes.func,
    product: PropTypes.object.isRequired
  }

  static defaultProps = {
    onChange: _.noop
  }

  constructor({ onChange, product }) {
    super()

    this.onChange = onChange
    this.resourceRef = React.createRef();

    this.state = {
      bookingOrder: false,
      order: null,
      savingOrder: false
    }
  }

  componentDidMount() {
    const { product } = this.props
    const { occsn: { Product }, registerComponent } = this.context

    if(!(product.isA && product.isA(Product))) {
      throw TypeError('Order component received prop product not of type Product');
    }

    registerComponent('order', this)

    this.constructOrder()
  }

  allowedToBookOrder = () => {
    const { components: { missingAnswers }} = this.context
    const { bookingOrder, order, savingOrder } = this.state

    if (!order || !missingAnswers) return false

    return (
      !bookingOrder && !savingOrder && missingAnswers.missingRequiredAnswers(order).empty()
    )
  }

  validateOrderBookable = (order) => {
    const { paymentForm, redeemables } = this.context

    // If the user hits enter while focused on the redeemables input, then submit the
    // redeemable search and cancel the order form submissionm
    if (redeemables && redeemables.state.focused) {
      redeemables.checkForRedeemable(redeemables.state.code)
      throw order
    }

    if (!order.outstandingBalance.isZero()) {
      let balanceTransaction = order
        .transactions()
        .target()
        .detect(t => !(t.paymentMethod() && t.paymentMethod(occsn.GiftCard)))

      if (balanceTransaction)
        order
          .transactions()
          .target()
          .delete(balanceTransaction)

      return paymentForm.chargeOutstandingBalanceToPaymentMethod(order)
    } else {
      return order
    }
  }

  bookOrder = (order) => {
    this.setState({
      bookingOrder: true
    }, async () => {
      try {
        order = await order.update({ status: 'reserved' })
      } catch (invalidOrder) {
        order = invalidOrder
      }

      this.setOrder(order, () => this.setState({ bookingOrder: false }))
    })
  }

  constructOrder = () => {
    const { product } = this.props
    const { occsn: { Order }} = this.context

    Order.construct({ product, status: 'initialized' }).then(this.setOrder)
  }

  isBooking = () => {
    return this.state.bookingOrder
  }

  isSaving = () => {
    return this.state.savingOrder
  }

  getOrder = () => {
    return this.state.order
  }

  saveOrder = async (order) => {
    this.setState({
      savingOrder: true
    }, async () => {
      let newOrder = await order.save()

      this.setOrder(newOrder, () => this.setState({ savingOrder: false }))
    })
  }

  setOrder = (order, onSet = _.noop) => {
    this.setState({ order }, () => {
      onSet(order)
      this.onChange(order)
    })
  }

  // Used by bookOrder button to initiate form submission
  submit = () => {
    this.resourceRef.handleSubmit();
  }

  render() {
    const { children, className } = this.props
    const { bookingOrder, order, savingOrder } = this.state

    let classNames = classnames('occsn-order-form', className)

    return (
      <section className={classNames}>
        {order && (
          <Resource
            afterError={this.setOrder}
            afterUpdate={this.saveOrder}
            beforeSubmit={this.validateOrderBookable}
            onInvalidSubmit={this.setOrder}
            onSubmit={this.bookOrder}
            ref={this.resourceRef}
            subject={order}
          >
            {children}
          </Resource>
        )}
      </section>
    )
  }
}
