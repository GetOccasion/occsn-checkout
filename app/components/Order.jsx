import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

import Decimal from 'decimal.js-light';
import _ from 'underscore';

import { Resource } from 'mitragyna';
import { Button } from 'reactstrap';

import occsn from '../libs/Occasion';

import TimeSlotsContainer from '../containers/TimeSlotsContainer.jsx';

import Attendees from './Order/Attendees.jsx';
import Customer from './Order/Customer.jsx';
import MissingAnswers from './Order/MissingAnswers.jsx';
import OrderErrors from './Order/OrderErrors.jsx';
import Pricing from './Order/Pricing.jsx';
import PaymentForm from './Order/PaymentForm.jsx';
import Questions from './Order/Questions.jsx';
import Redeemables from './Order/Redeemables.jsx';

export default class Order extends PureComponent {
  static propTypes = {
    activeTimeSlotsCollection: PropTypes.shape({
      // __collection: PropTypes.arrayOf(PropTypes.instanceOf(occsn.TimeSlot))
      __collection: PropTypes.array
    }),
    timeSlotsFromCalendar: PropTypes.shape({
      __collection: PropTypes.arrayOf(PropTypes.instanceOf(occsn.TimeSlot))
    }),
    afterError: PropTypes.func.isRequired,
    afterUpdate: PropTypes.func.isRequired,
    bookingOrder: PropTypes.bool,
    findRedeemable: PropTypes.func.isRequired,
    onSubmit: PropTypes.func,
    saveOrder: PropTypes.func,
    savingOrder: PropTypes.bool,
    setSkipAttendee: PropTypes.func,
    skipAttendees: PropTypes.object,
    subject: PropTypes.instanceOf(occsn.Order).isRequired,
    spreedlyIframeInputStyles:  PropTypes.object,
    squareIframeInputStyles:  PropTypes.object,
  };

  static contextTypes = {
    callbackProps: PropTypes.object,
    componentProps: PropTypes.object,
  };

  constructor() {
    super();
    _.bindAll(this,
      'allowedToBookOrder',
      'headerForSection',
      'showPaymentForm',
      'showPrice',
    );
  }

  allowedToBookOrder() {
    const { bookingOrder, savingOrder, subject } = this.props;

    if(!subject || !this.missingAnswers) return false;

    return !bookingOrder && !savingOrder && this.missingAnswers.missingRequiredAnswers(subject).empty();
  }

  // Mitragyna callback
  beforeSubmit(subject) {
    if(this.redeemables && this.redeemables.state.focused) {
      this.redeemables.checkForRedeemable(this.redeemables.state.code);
      throw subject;
    }

    if(!subject.outstandingBalance.isZero()) {
      let balanceTransaction =
        subject.transactions().target().detect((t) => !(t.paymentMethod() && t.paymentMethod(occsn.GiftCard)));

      if(balanceTransaction) subject.transactions().target().delete(balanceTransaction);

      return this.paymentForm.chargeOutstandingBalanceToPaymentMethod(subject);
    } else {
      return subject;
    }
  }

  // @param [String] section the name of the section to get the custom or default section title for, rendered as <h2>
  headerForSection(section) {
    const { subject } = this.props;
    let product = subject.product();

    switch(section) {
      case 'contact':
        return <div className="container-title" id="widgetContactTitle">
          <h4>
            { _.isNull(product.widgetContactTitle) ? (
              "Customer Information"
            ) : (
              product.widgetContactTitle
            )}
          </h4>
          <hr className="pb-3"/>
        </div>;
      case 'timeSlots':
        return <div className="container-title" id="widgetTimeSlotsTitle">
          <h4>
            { _.isNull(product.widgetTimeSlotsTitle) ? (
              "Select which day you would like to reserve"
            ) : (
              product.widgetTimeSlotsTitle
            )}
          </h4>
          <hr className="pb-3"/>
        </div>;
      case 'questions':
        return <div className="container-title" id="widgetQuestionsTitle">
          <h4>
            { _.isNull(product.widgetQuestionsTitle) ? (
              "Additional Questions"
            ) : (
              product.widgetQuestionsTitle
            )}
          </h4>
          <hr className="pb-3"/>
        </div>;
      case 'attendees':
        return <div className="container-title" id="widgetAttendeesTitle">
          <h4>Attendee Information</h4>
          <hr className="pb-3"/>
        </div>;
      case 'payment':
        return <div className="container-title" id="widgetPaymentTitle">
          <h4>
            { _.isNull(product.widgetPaymentTitle) ? (
              "Payment Information"
            ) : (
              product.widgetPaymentTitle
            )}
          </h4>
          <hr className="pb-3"/>
        </div>;
      case 'redeemables':
        return <div className="container-title" id="widgetRedeemableTitle">
            <h4>Coupons and Gift Cards</h4>
            <hr className="pb-3"/>
        </div>;
      case 'totalDue':
        return <div className="container-title" id="widgetTotalDueTitle">
          <h4>
            { _.isNull(product.widgetTotalDueTitle) ? (
              (product.depositMethod == 'no_deposit') ? "Total Due Today" : "Deposit Due Today"
            ) : (
              product.widgetTotalDueTitle
            )}
          </h4>
          <hr className="pb-3"/>
        </div>;
    }
  }

  // Determines if should show Attendees
  // @note If the product is quantity aware && has attendeeQuestions, and order.quantity > 0, show Attendees
  //
  // @return [Boolean] whether or not to show Attendees
  showAttendees() {
    let { subject } = this.props;
    let product = subject.product();

    if(subject.newResource()) return false;

    return !product.attendeeQuestions.empty() && subject.quantity > 0;
  }

  // Determines if should show TimeSlotsContainer
  // @note If the product has firstTimeSlotStartsAt and requiresTimeSlotSelection, and a timeSlot has not been
  //   pre-selected via window.OCCSN.time_slot_id, then show TimeSlotsContainer
  //
  // @return [Boolean] whether or not to show TimeSlotsContainer
  showTimeSlots() {
    let { subject } = this.props;
    let product = subject.product();

    return product.firstTimeSlotStartsAt && (product.requiresTimeSlotSelection || product.sellsSessions) && !window.OCCSN.time_slot_id;
  }

  // Determines if should show Questions
  // @note If the product.questions() is empty, don't show questions
  //
  // @return [Boolean] whether or not to show Questions
  showQuestions() {
    let { subject } = this.props;
    let product = subject.product();

    return !product.questions().empty();
  }

  // Determines if should show Redeemables
  // @note If the product is not free && hasRedeemables then show Redeemables
  //
  // @return [Boolean] whether or not to show Redeemables
  showRedeemables() {
    let { subject } = this.props;
    let product = subject.product();

    if(subject.newResource()) return false;

    return !product.free && product.hasRedeemables && ((subject.subtotal && !subject.subtotal.isZero()) || window.OCCSN.coupon_code);
  }

  // Determines if should show the payment form
  // @note If the product is free or outstandingBalance is zero, do not show the payment form
  //
  // @return [Boolean] whether or not to show the payment form
  showPaymentForm() {
    let { subject } = this.props;
    let product = subject.product();

    if(subject.newResource()) return false;

    return !(product.free || !subject.outstandingBalance || subject.outstandingBalance.isZero());
  }

  // Determines if should show the price output
  // @note If the product is free or price is null (required price-calculating questions are missing answers),
  //   do not show the price display
  //
  // @return [Boolean] whether or not to show the payment form
  showPrice() {
    let { subject } = this.props;
    let product = subject.product();

    if(subject.newResource()) return false;

    return !product.free && subject.price;
  }

  renderBookOrderButton = () => {
    const { bookingOrder, onSubmit, subject } = this.props;
    const { callbackProps, componentProps } = this.context;

    let button = <Button
      block
      className="book-order-button"
      color="success"
      id="bookOrder"
      size="lg"
      disabled={ !this.allowedToBookOrder() }
      onClick={onSubmit}
    >
      <span>{ subject.product().orderButtonText }</span>
      {
        bookingOrder && componentProps.orderBooking && (
          React.createElement(componentProps.orderBooking)
        )
      }
    </Button>;

    if(callbackProps.onBookOrderButtonRender) callbackProps.onBookOrderButtonRender(button);

    return button;
  };

  render() {
    let { afterError, saveOrder, findRedeemable, setSkipAttendee, skipAttendees, subject, spreedlyIframeInputStyles, squareIframeInputStyles } = this.props;

    let customer = subject.customer();
    let product = subject.product();

    return <section className="order-container">
      {
        this.showTimeSlots() && (
          <section className="time-slots-container" id="time-slots-container">
            <a name="time-slots" id="time-slots-anchor"></a>
            { this.headerForSection('timeSlots') }
            <TimeSlotsContainer order={subject} />
          </section>
        )
      }

      <section className="information-container" id="information-container">
        <section className="customer-container" id="customer-container">
          <a name="customer" id="customer-anchor"></a>
          { this.headerForSection('contact') }
          <Resource component={ Customer } reflection="customer" subject={ customer }></Resource>
        </section>

        {
          this.showQuestions() && (
            <section className="questions-container" id="questions-container">
              <a name="questions" id="questions-anchor"></a>
              { this.headerForSection('questions') }
              <Questions subject={subject} questions={ product.questions().target() }></Questions>
            </section>
          )
        }

        {
          this.showAttendees() && (
            <section className="attendees-container" id="attendees-container">
              <a name="attendees" id="attendees-anchor"></a>
              { this.headerForSection('attendees') }
              <Attendees questions={product.attendeeQuestions} setSkipAttendee={setSkipAttendee} skipAttendees={skipAttendees} subject={subject}></Attendees>
            </section>
          )
        }
      </section>

      <section className="payments-container" id="payments-container">
        {
          this.showRedeemables() && (
            <section className="redeemables-container" id="redeemables-container">
              <a name="redeemables" id="redeemables-anchor"></a>
              { this.headerForSection('redeemables') }
              <Redeemables
                findRedeemable={findRedeemable}
                order={subject}
                onChange={saveOrder}
                onErrors={afterError}
                ref={(r) => this.redeemables = r}
              ></Redeemables>
            </section>
          )
        }

        {
          this.showPaymentForm() && (
            <section className="payment-container" id="payment-container">
              <a name="payment" id="payment-anchor"></a>
              { this.headerForSection('payment') }
              <PaymentForm
                order={subject}
                spreedlyIframeInputStyles={spreedlyIframeInputStyles}
                squareIframeInputStyles={squareIframeInputStyles}
                ref={(form) => this.paymentForm = form }
              ></PaymentForm>
            </section>
          )
        }

        {
          this.showPrice() && (
            <section className="total-due-container" id="total-due-container">
              <a name="total-due" id="total-due-anchor"></a>
              { this.headerForSection('totalDue') }
              <Pricing order={subject}></Pricing>
            </section>
          )
        }
      </section>

      <section className="missing-answers-container" id="missing-answers-container">
        <a name="missing-answers"></a>
        <MissingAnswers order={subject} ref={(r) => this.missingAnswers = r }></MissingAnswers>
      </section>

      <section className="order-errors-container" id="order-errors-container">
        <a name="order-errors"></a>
        <OrderErrors order={subject}></OrderErrors>
      </section>

      <section className="book-order-container" id="book-order-container">
        { this.renderBookOrderButton() }
      </section>
    </section>;
  }
}
