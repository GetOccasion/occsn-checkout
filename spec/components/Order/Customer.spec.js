import React from 'react';

import ActiveResource from 'active-resource';
import { Resource } from 'mitragyna';

import Order from '../../../app/components/Order';
import Customer from '../../../app/components/Order/Customer';

import axios from 'axios';
import occsn from '../../../app/libs/Occasion';

import productFixture from 'fixtures/products/cash.json';
import blankQuestionsFixture from 'fixtures/blank.json';

import initializedOrderFixture from 'fixtures/orders/initialized/free.json';
import productTimeSlotsFixture from 'fixtures/products/time_slots.json';
import partialCustomerOrderFixture from 'fixtures/orders/customer/partial.json';
import completeCustomerOrderFixture from 'fixtures/orders/customer/complete.json';
import customerOrderErrorsFixture from 'fixtures/errors/orders/customer.json';

describe('Order', () => {
  describe('Customer', () => {
    let order, product, wrapper, customerWrapper;

    const mockSetOrder = jest.fn();

    async function setupWrapper(orderResponses) {
      let responses = {
        '/products/:id/': { status: 200, data: productFixture },
        '/products/:id/questions/': { status: 200, data: blankQuestionsFixture },
        '/products/:id/time_slots/': { status: 200, data: blankQuestionsFixture },
        ...orderResponses
      };
      axios._setMockResponses(responses);

      product = await occsn.Product.find('1');
      order = await occsn.Order.construct({ product, status: 'initialized' });

      try {
        order = await order.save();
      } catch(invalidOrder) {
        order = invalidOrder;
      }

      let props = {
        component: Order,
        componentProps: { selectedTimeSlots: ActiveResource.Collection.build() },
        subject: order,
        afterUpdate: mockSetOrder
      };

      wrapper = mount(<Resource {...props} />);
      customerWrapper = wrapper.find(Customer).first();
    }

    context('on change', () => {
      beforeEach(async () => {
        await setupWrapper({
          '/orders/': { status: 201, data: initializedOrderFixture },
          '/orders/:id': { status: 200, data: partialCustomerOrderFixture },
        });

        var emailInput = customerWrapper.find('input').first();
        emailInput.simulate('change', { target: { value: 'nick@example.com' } });
        emailInput.simulate('blur');

        await axios.wait(1000);
      });

      it('calls setOrder', () => {
        expect(mockSetOrder.mock.calls.length).toBe(1);
      });

      it('matches snapshot', () => {
        wrapper.update();
        expect(wrapper.find(Customer).first().find('section').first()).toMatchSnapshot();
      });
    });

    context('complete customer', () => {
      beforeEach(async () => {
        await setupWrapper({
          '/orders/': { status: 201, data: completeCustomerOrderFixture },
          '/orders/:id': { status: 200, data: completeCustomerOrderFixture },
        });
      });

      it('matches snapshot', () => {
        expect(customerWrapper.find('section').first()).toMatchSnapshot();
      });
    });

    context('customer errors', () => {
      beforeEach(async () => {
        await setupWrapper({
          '/orders/': { status: 422, data: customerOrderErrorsFixture },
        });
      });

      it('matches snapshot', () => {
        expect(customerWrapper.find('section').first()).toMatchSnapshot();
      });
    });
  });
});
