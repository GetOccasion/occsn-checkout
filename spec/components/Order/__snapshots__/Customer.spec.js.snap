// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order Customer complete customer matches snapshot 1`] = `
<section>
  <ListGroup
    tag="ul"
  >
    <ul
      className="list-group"
    >
      <ListGroupItem
        tag="li"
      >
        <li
          className="list-group-item"
        >
          <Field
            component={[Function]}
            invalidClassName="is-invalid"
            name="email"
            placeholder="Email"
            type="email"
          >
            <Input
              className=""
              key="email"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="Email"
              type="email"
              value="nick@example.com"
            >
              <input
                className="form-control"
                onBlur={[Function]}
                onChange={[Function]}
                placeholder="Email"
                type="email"
                value="nick@example.com"
              />
            </Input>
          </Field>
          <ErrorsFor
            component={[Function]}
            field="email"
          />
        </li>
      </ListGroupItem>
      <ListGroupItem
        tag="li"
      >
        <li
          className="list-group-item"
        >
          <Field
            component={[Function]}
            invalidClassName="is-invalid"
            name="firstName"
            placeholder="First Name"
            type="text"
          >
            <Input
              className=""
              key="firstName"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="First Name"
              type="text"
              value="Nick"
            >
              <input
                className="form-control"
                onBlur={[Function]}
                onChange={[Function]}
                placeholder="First Name"
                type="text"
                value="Nick"
              />
            </Input>
          </Field>
          <ErrorsFor
            component={[Function]}
            field="firstName"
          />
        </li>
      </ListGroupItem>
      <ListGroupItem
        tag="li"
      >
        <li
          className="list-group-item"
        >
          <Field
            component={[Function]}
            invalidClassName="is-invalid"
            name="lastName"
            placeholder="Last Name"
            type="text"
          >
            <Input
              className=""
              key="lastName"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="Last Name"
              type="text"
              value="Smith"
            >
              <input
                className="form-control"
                onBlur={[Function]}
                onChange={[Function]}
                placeholder="Last Name"
                type="text"
                value="Smith"
              />
            </Input>
          </Field>
          <ErrorsFor
            component={[Function]}
            field="lastName"
          />
        </li>
      </ListGroupItem>
      <ListGroupItem
        tag="li"
      >
        <li
          className="list-group-item"
        >
          <Field
            component={[Function]}
            invalidClassName="is-invalid"
            name="zip"
            placeholder="Zip Code"
            type="text"
          >
            <Input
              className=""
              key="zip"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="Zip Code"
              type="text"
              value=""
            >
              <input
                className="form-control"
                onBlur={[Function]}
                onChange={[Function]}
                placeholder="Zip Code"
                type="text"
                value=""
              />
            </Input>
          </Field>
          <ErrorsFor
            component={[Function]}
            field="zip"
          />
        </li>
      </ListGroupItem>
    </ul>
  </ListGroup>
</section>
`;

exports[`Order Customer customer errors matches snapshot 1`] = `
<section>
  <ListGroup
    tag="ul"
  >
    <ul
      className="list-group"
    >
      <ListGroupItem
        tag="li"
      >
        <li
          className="list-group-item"
        >
          <Field
            component={[Function]}
            invalidClassName="is-invalid"
            name="email"
            placeholder="Email"
            type="email"
          >
            <Input
              className="is-invalid"
              key="email"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="Email"
              type="email"
              value=""
            >
              <input
                className="is-invalid form-control"
                onBlur={[Function]}
                onChange={[Function]}
                placeholder="Email"
                type="email"
                value=""
              />
            </Input>
          </Field>
          <ErrorsFor
            component={[Function]}
            field="email"
          >
            <FormFeedback
              key="email"
              tag="div"
            >
              <div
                className="invalid-feedback"
              >
                <span
                  key="blank"
                >
                  Email cannot be blank.
                </span>
              </div>
            </FormFeedback>
          </ErrorsFor>
        </li>
      </ListGroupItem>
      <ListGroupItem
        tag="li"
      >
        <li
          className="list-group-item"
        >
          <Field
            component={[Function]}
            invalidClassName="is-invalid"
            name="firstName"
            placeholder="First Name"
            type="text"
          >
            <Input
              className="is-invalid"
              key="firstName"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="First Name"
              type="text"
              value=""
            >
              <input
                className="is-invalid form-control"
                onBlur={[Function]}
                onChange={[Function]}
                placeholder="First Name"
                type="text"
                value=""
              />
            </Input>
          </Field>
          <ErrorsFor
            component={[Function]}
            field="firstName"
          >
            <FormFeedback
              key="firstName"
              tag="div"
            >
              <div
                className="invalid-feedback"
              >
                <span
                  key="blank"
                >
                  First Name cannot be blank.
                </span>
              </div>
            </FormFeedback>
          </ErrorsFor>
        </li>
      </ListGroupItem>
      <ListGroupItem
        tag="li"
      >
        <li
          className="list-group-item"
        >
          <Field
            component={[Function]}
            invalidClassName="is-invalid"
            name="lastName"
            placeholder="Last Name"
            type="text"
          >
            <Input
              className="is-invalid"
              key="lastName"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="Last Name"
              type="text"
              value=""
            >
              <input
                className="is-invalid form-control"
                onBlur={[Function]}
                onChange={[Function]}
                placeholder="Last Name"
                type="text"
                value=""
              />
            </Input>
          </Field>
          <ErrorsFor
            component={[Function]}
            field="lastName"
          >
            <FormFeedback
              key="lastName"
              tag="div"
            >
              <div
                className="invalid-feedback"
              >
                <span
                  key="blank"
                >
                  Last Name cannot be blank.
                </span>
              </div>
            </FormFeedback>
          </ErrorsFor>
        </li>
      </ListGroupItem>
      <ListGroupItem
        tag="li"
      >
        <li
          className="list-group-item"
        >
          <Field
            component={[Function]}
            invalidClassName="is-invalid"
            name="zip"
            placeholder="Zip Code"
            type="text"
          >
            <Input
              className=""
              key="zip"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="Zip Code"
              type="text"
              value=""
            >
              <input
                className="form-control"
                onBlur={[Function]}
                onChange={[Function]}
                placeholder="Zip Code"
                type="text"
                value=""
              />
            </Input>
          </Field>
          <ErrorsFor
            component={[Function]}
            field="zip"
          />
        </li>
      </ListGroupItem>
    </ul>
  </ListGroup>
</section>
`;

exports[`Order Customer on change matches snapshot 1`] = `
<section>
  <ListGroup
    tag="ul"
  >
    <ul
      className="list-group"
    >
      <ListGroupItem
        tag="li"
      >
        <li
          className="list-group-item"
        >
          <Field
            component={[Function]}
            invalidClassName="is-invalid"
            name="email"
            placeholder="Email"
            type="email"
          >
            <Input
              className=""
              key="email"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="Email"
              type="email"
              value="nick@example.com"
            >
              <input
                className="form-control"
                onBlur={[Function]}
                onChange={[Function]}
                placeholder="Email"
                type="email"
                value="nick@example.com"
              />
            </Input>
          </Field>
          <ErrorsFor
            component={[Function]}
            field="email"
          />
        </li>
      </ListGroupItem>
      <ListGroupItem
        tag="li"
      >
        <li
          className="list-group-item"
        >
          <Field
            component={[Function]}
            invalidClassName="is-invalid"
            name="firstName"
            placeholder="First Name"
            type="text"
          >
            <Input
              className=""
              key="firstName"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="First Name"
              type="text"
              value=""
            >
              <input
                className="form-control"
                onBlur={[Function]}
                onChange={[Function]}
                placeholder="First Name"
                type="text"
                value=""
              />
            </Input>
          </Field>
          <ErrorsFor
            component={[Function]}
            field="firstName"
          />
        </li>
      </ListGroupItem>
      <ListGroupItem
        tag="li"
      >
        <li
          className="list-group-item"
        >
          <Field
            component={[Function]}
            invalidClassName="is-invalid"
            name="lastName"
            placeholder="Last Name"
            type="text"
          >
            <Input
              className=""
              key="lastName"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="Last Name"
              type="text"
              value=""
            >
              <input
                className="form-control"
                onBlur={[Function]}
                onChange={[Function]}
                placeholder="Last Name"
                type="text"
                value=""
              />
            </Input>
          </Field>
          <ErrorsFor
            component={[Function]}
            field="lastName"
          />
        </li>
      </ListGroupItem>
      <ListGroupItem
        tag="li"
      >
        <li
          className="list-group-item"
        >
          <Field
            component={[Function]}
            invalidClassName="is-invalid"
            name="zip"
            placeholder="Zip Code"
            type="text"
          >
            <Input
              className=""
              key="zip"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="Zip Code"
              type="text"
              value=""
            >
              <input
                className="form-control"
                onBlur={[Function]}
                onChange={[Function]}
                placeholder="Zip Code"
                type="text"
                value=""
              />
            </Input>
          </Field>
          <ErrorsFor
            component={[Function]}
            field="zip"
          />
        </li>
      </ListGroupItem>
    </ul>
  </ListGroup>
</section>
`;
